catalog/service/impl/src/main/java/org/projectnessie/catalog/service/impl/ImportSnapshotWorker.java

```diff
@@ -109,7 +109,7 @@ private EntitySnapshotObj.Builder importIcebergTable(
       StorageUri metadataLocation = StorageUri.of(content.getMetadataLocation());
       try {
         InputStream input = taskRequest.objectIO().readObject(metadataLocation);
-        if (metadataLocation.requiredPath().endsWith(".gz")) {
+        if (metadataLocation.requiredPath().endsWith("metadata.json.gz") || metadataLocation.requiredPath().endsWith(".gz.metadata.json")) {
```

Comment:
Would you mind keeping the first case more generic as `.endsWith(".gz")` to be consistent with old behaviour?

Could you add a test for the second case?

Reply:
Maybe in `AbstractIcebergCatalogTests`?